src/map/utils/battleutils.cpp

```diff
@@ -4098,38 +4042,118 @@ namespace battleutils
+                        { // mob is currently claimed by your alliance, update ownership
+                            mob->m_OwnerID.id = PAttacker->id;
+                            mob->m_OwnerID.targid = PAttacker->targid;
+                            if (PDefender->isAlive())
+                            { // ignore killing blow
+                                mob->updatemask |= UPDATE_STATUS;
+                                attacker->PClaimedMob = PDefender;
+                            }
+                        }
+                        else
+                        { // mob is unclaimed
+                            PAttacker->ForAlliance([&PAttacker, &PDefender, &mob, &attacker](CBattleEntity* PMember){
+                                if (mob->PEnmityContainer->GetHighestEnmity() == PMember || mob->PEnmityContainer->GetHighestEnmity() == PMember->PPet)
+                                { // someone in your alliance is top of hate list, claim for your alliance
+                                    mob->m_OwnerID.id = PAttacker->id;
+                                    mob->m_OwnerID.targid = PAttacker->targid;
+                                    if (PDefender->isAlive())
+                                    { // ignore killing blow
+                                        mob->updatemask |= UPDATE_STATUS;
+                                        attacker->PClaimedMob = PDefender;
+                                    }
+                                }
+                            });
+                        }
+                    }
                 }
+            }
+        }
+    }
 
-                if (!mob->CalledForHelp())
+    void DirtyExp(CBattleEntity* PDefender, CBattleEntity* PAttacker)
+    {
+        if (PDefender->objtype == TYPE_MOB)
+        {
+            CMobEntity* mob = static_cast<CMobEntity*>(PDefender);
+            if (PAttacker->objtype != TYPE_PC)
+            {
+                if (PAttacker->PMaster && PAttacker->PMaster->objtype == TYPE_PC)
+                {
+                    PAttacker = PAttacker->PMaster;
+                }
+                else
                 {
-                    mob->m_OwnerID.id = PAttacker->id;
-                    mob->m_OwnerID.targid = PAttacker->targid;
+                    PAttacker = nullptr;
                 }
-                mob->updatemask |= UPDATE_STATUS;
+            }
+            if (PAttacker)
+            {
+                uint8 pcinzone = 0;
+                PAttacker->ForAlliance([&pcinzone, &mob](CBattleEntity* PMember) {
+                    if (PMember->getZone() == mob->getZone() && distance(PMember->loc.p, mob->loc.p) < 100)
+                    {
+                        pcinzone++;
+                    }
+                });
+                mob->m_HiPartySize = std::max(pcinzone, mob->m_HiPartySize);
+                mob->m_HiPCLvl = std::max(PAttacker->GetMLevel(), mob->m_HiPCLvl);
```

Comment:
exp changes should be decided by highest player character to take action on or otherwise effect a mob directly - debuffs, damage (inc damage from spikes), target of JA - but not from actions that have no effect on the mob itself such as curing people from outside their party.

Reply:
Sorry, I should have been more clear; a high level WHM _in_ the party.

Reply:
The highest level party member is checked in dirtyExp, and also in charutils::DistributeExperiencePoints. So If a high level WHM doesn't physically interact with a mob, they should still be considered when exp is actually split. [Starting around here.](https://github.com/project-topaz/topaz/blob/432058d0ae232b2f0341b1e379e0f9b4a9515eb8/src/map/utils/charutils.cpp#L3265)

Reply:
Right; but the highest player level is a comparison against the attacker and the currently highest set for the mob:
`mob->m_HiPCLvl = std::max(PAttacker->GetMLevel(), mob->m_HiPCLvl);`
`dirtyExp` does loop through the entire party when a claiming action has been taken, but only to count the number of players in the party; as written it's not keeping track of the highest level player:
```
PAttacker->ForAlliance([&pcinzone, &mob](CBattleEntity* PMember) {
    if (PMember->getZone() == mob->getZone() && distance(PMember->loc.p, mob->loc.p) < 100)
    {
        pcinzone++;
    }
});
```
`DistributeExperiencePoints` does do so, but that only factors the WHM in upon mob death if the WHM or buffer remains in the party.

Reply:
Is the WHM supposed to factor into the party level if they don't act on the mob directly and they're not in the party when the mob dies? It's easy to change but I was under the impression that a high level WHM could be in the party during a fight, drop party, and not affect exp other than party size, given they don't interact with the mob.

From wiki:
`Party level is the level of the highest level player or pet to have acted on the monster since it was at 100% health with no debilitating status effects.`

Reply:
I _assumed_ it's supposed to affect exp but should probably have someone confirm~

@project-topaz/researcher

Reply:
just to clarify, the scenario you wish to test is:
player a lvl 50
player b lvl 75
in party together, engage mob, player b takes no action
player b drops party and either:
does nothing (what is xp?)
PLs (what is xp?)

Reply:
Player A can be level 1, Player B can be 99, as long as there's a sufficient difference in level to impact exp. Then party up, have only Player A attack the mob, and have Player B drop party before the mob dies. What does the exp look like when the mob dies? Does anything change if Player B cures/buffs Player A before dropping party?

Reply:
> but that only factors the WHM in upon mob death if the WHM or buffer remains in the party.

thats expected: on retail I've jumped in party power buffed ppl then left without effecting the exp. (haste n refresh) tho SE nerfed the buff durations)